### DISCUSSION TITLE
S-N curve data and loads file

### DISCUSSION THREAD

**Discussion by lt72884** [OpenFAST/openfast#2034] [Q&A]
Ok, so i wrote a nice project paper using openFAST and MLife. I was asked to verify the data using ANSYS and other software( have not decided yet) to verify the results from openfast and mlife. 

I was asked to get the S-N curve for the 5mw DLL land wind turb that i have been using. 

1.Does this curve exist?
2.Can MLife produce a S-N curve?

Next question, I am testing out a opensource software called FATlab and it needs a "loads file" which contains the time series, cyclic loading of whatever your testing. 

1.Is there anyway to get that from openFAST? It has to be in a xls format, if i recall :)
2. The blades are made from fiberglass, do we know the youngs mod and the poisons ratio?

Thank you much

**Comment by jjonkman** [comment]
Dear @lt72884,

Regarding the S-N curve in MLife, see a similar question asked and answered in the following topic on our forum: https://forums.nrel.gov/t/derivation-of-del-equation/5514.

I'm not sure I understand your second set of questions regarding FatLab.  OpenFAST calculates time-series of section loads and MLife performs the rainflow cycle counting to count cycles of load ranges, damage-equivalent loads, and fatigue.

Best regards,

**Comment by lt72884** [comment]
yeah, the second question regarding FATlab is a little odd, i agree, but thats what fatlab wants. Ill answer that in a second, but first, is there any data on the material of the blade? i have looked in the "definition of a 5MW" pdf and cant find anything. All i literally need is the youngs modulus and poisons ratio. 

So according to the documentation of fat lab, the loads file is a xls file that has number of repititons (n), time, load case at that time for Fx, Fy, and Mz. 

its like the "RootFxb1" "RootFyb1" and "RootMzb1" but all combined is my guess. Here is a screen shot from the documentation.  Thank you much for all your help these last 7 months. I have made sure to acknowledge your patience and efforts in all my projects.

![loads file](https://github.com/OpenFAST/openfast/assets/52751207/ef0276ac-02c1-4f17-a199-892d550fced8)

**Reply by jjonkman** [reply]
Dear @lt72884,

If I understand correctly, "n" in this spreadsheet is the number of time steps and you are specifying the time series of loads, including force in x and y and moment about z.  Such load time series can be computed directly from OpenFAST (with no need to run MLife).

Best regards,

**Comment by lt72884** [comment]
Im not sure if n is time steps. The reason i say that, the scrolling bar on the right is way to big to have 300,000 time stamps in that file.

I do have all those loads calculated from openfast, just not sure how to export them to excel... I think pydatview can do it. Ill test it

Ok, what about the materials question regarding youngs modulus, poison ratio and an sn curve since mlife doesnt generate one:)

Thanks

**Reply by jjonkman** [reply]
Regarding "n", that is a better question for FatLab technical support.

The OpenFAST output could be copied/pasted into Excel (if you only need to do this for a few cases), or you could write a script to generate the Excel spreadsheet from the OpenFAST output.

The NREL 5-MW baseline wind turbine specifications report does not provide the material properties of the blade.  In reality, a blade of this size would be made of composite materials rather than isotropic materials.  But if this is useful to you, the composite layup of the NREL 5-MW baseline blade has been reversed engineered by Brian Resor of Sandia National Laboratory, as discussed in the following  topic on our forum: https://forums.nrel.gov/t/structural-design-of-the-wind-turbine-blade/271.

Best regards,

**Comment by lt72884** [comment]
i agree, contacting fatlab would be best. I have been trying to do so today :) 

Ill read the link thanks

Its just been VERY hard trying to find an sn curve to use in ansys. Im still new to ansys, and i have been asked to get a sn curve for the material... so hard to find one hahaha

thanks for all the help so far

**Comment by lt72884** [comment]
do you by chance have the PDF since it requries authintication now as per your reply on the forum. I also tested it and cant access it. 

""Are you asking for the composite layup of the NREL 5-MW baseline wind turbine blade that was reversed engineered by Brian Resor of SNL? This report has been discussed here: [Structural design of the wind turbine blade 8](https://forums.nrel.gov/t/structural-design-of-the-wind-turbine-blade/271), but apparently the report is no longer available online (without authorization). I can send an electronic copy if that is what you want.

Best regards,""

**Reply by jjonkman** [reply]
I wont upload it to a repository, but I could e-mail it to you if you send me your address.

Best regards,

**Reply by lt72884** [reply]
yes please. Send to LT72884@gmail.com

if you have any of the blade files from him, and whatever else, that would be great. thank you

**Reply by jjonkman** [reply]
Sent.

**Comment by lt72884** [comment]
thank you:)

**Comment by lt72884** [comment]
I noticed on the excel output from MLife that it says "Basic Time Until Failure (s) for various S/N Curves" where is MLife getting this sn curve data?

thanks

**Reply by jjonkman** [reply]
Dear @lt72884,

The S/N curve is user-specified in MLife via inputs `SNslopeLst` (the Wholer exponent) and `LUlt` (the y- (S-) axis intercept).  You can specify more than one Wholer exponent (hence the word "various") by setting `NSlopes` > 1 (in which case you should specify multiple values in `SNslopeLst`).

Best regards,

**Reply by lt72884** [reply]
interesting, from these 2 values, can i create an SN curve to give to ansys? If i remember correctly,, 10 is my wholer exponent

![image](https://github.com/OpenFAST/openfast/assets/52751207/7b0efead-2af0-467d-8cfd-b8a15b1f40f0)

**Reply by jjonkman** [reply]
I'm not familiar with the needs of ANSYS, but I see your Wholer exponent (m=10) and `L_Ult` values in your table, which are sufficient for defining an S/N curve.

Best regards,

**Comment by lt72884** [comment]
Ok, maybe i can create one. yeah, ansys needs a full sn curve data. So you add cycles as the x axis and stress as the y axis. But if i can make my own curve from this information that would be great. I just need to remember how to do that haha. 

for MLife L_ult is the breaking force? or how is LUlt defined? 

Also, dont i need an enduance limit?

thanks

**Reply by jjonkman** [reply]
The S/N curve used by MLife is documented in Eq. (2) of the MLife Theory Manual: https://www.nrel.gov/wind/nwtc/assets/pdfs/mlife-theory.pdf.

Best regards,

**Reply by lt72884** [reply]
interesting. So it doesnt use an endurance limit. Ill see if i can somehow graph that equation using the information i have.

Where can i find the LRF value? and which LMF value am i using in the equation 2? and which LMF correlates to my "basic time until failure" image above. It only shows LULT and m, but no LMF.. would it be 0? Also  what is the nref number (161596703.887162) and how does that work?

![image](https://github.com/OpenFAST/openfast/assets/52751207/dc7e86c7-b4c6-466b-80e1-79bd615e4837)

What Lmf and Lrf value for this image below: )

![image](https://github.com/OpenFAST/openfast/assets/52751207/be548924-4627-45c3-8220-b92fb3a0c671)
